src/frequenz/sdk/microgrid/component_graph.py

```diff
@@ -853,10 +853,31 @@ def find_first_descendant_component(
             ValueError: When the root component is not found in the component
                 graph or when no component is found in the given categories.
         """
-        root_component = next(
-            (comp for comp in self.components(component_categories={root_category})),
-            None,
-        )
+        if root_category == ComponentCategory.GRID:
+            root_component = next(
+                (
+                    comp
+                    for comp in self.components(component_categories={root_category})
+                ),
+                None,
+            )
+        else:
+            grid_component = next(
+                (
+                    comp
+                    for comp in self.components(
+                        component_categories={ComponentCategory.GRID}
+                    )
+                ),
+                None,
+            )
+            if grid_component is None:
+                raise ValueError("Grid component not found in component graph!")
```

Comment:
I would raise a `InvalidGraphError` as there is really nothing wrong with the values passed to the function, is just the component graph is not suitable for this operation, right?
```suggestion
                raise InvalidGraphError("Grid component not found in component graph!")
```

Reply:
done